spotify_player/src/cli/client.rs

```diff
@@ -350,3 +365,453 @@ async fn handle_playback_request(
+                        .unwrap();
+
+                    // No import for non following playlist
+                    if pl_follow {
+                        let to_dir = imports_dir.join(dir_name);
+                        for file in to_dir.read_dir()? {
+                            let file_name = file?.file_name().into_string().unwrap();
+
+                            let from_id = PlaylistId::from_id(file_name)?;
+
+                            result.push_str(
+                                playlist_import(client, from_id, to_id.to_owned(), delete)
+                                    .await?
+                                    .as_str(),
+                            );
+                            result.push('\n');
+                        }
+                    } else {
+                        // Remove non following imports as they are now useless
+                        remove_dir_all(dir_path)?;
+                        result.push_str(
+                            format!(
+                                "Not following playlist '{}'. Deleting import...\n",
+                                to_id.id()
+                            )
+                            .as_str(),
+                        )
+                    }
+                }
+
+                Ok(result)
+            }
+        }
+    }
+}
+
+/// Imports a playlist into another playlist.
+/// All tracks from the imported playlist are added to the import-to playlist if they are not in there already.
+/// The state of the imported playlist and a hash of the track ids are written to a file
+///
+/// This file is used in subsequent imports of the same two playlists
+/// It is first used to see if the state of the imported playlist has changed
+/// then if so, is used to see what has differed in the change.
+async fn playlist_import(
+    client: &Client,
+    import_from: PlaylistId<'static>,
+    import_to: PlaylistId<'static>,
+    delete: bool,
+) -> Result<String> {
+    #[derive(PartialEq, Eq, Hash)]
+    struct PlaylistData {
+        id: TrackId<'static>,
+        name: String,
+    }
+
+    // Get playlists' info, checking if they exist
+    let (from, from_tracks) = match client.playlist_context(import_from.to_owned()).await? {
+        Context::Playlist { playlist, tracks } => (playlist, tracks),
+        _ => panic!(
+            "Cannot import from {}. Playlist not found.",
+            import_from.id()
+        ),
+    };
+
+    let (to, to_tracks) = match client.playlist_context(import_to.to_owned()).await? {
+        Context::Playlist { playlist, tracks } => (playlist, tracks),
+        _ => panic!("Cannot import to {}. Playlist not found.", import_to.id()),
+    };
+
+    // Create hashset of import_to playlist
+    // Will use to check before adding track
+    let mut to_hash = HashSet::new();
+    for track in to_tracks {
+        to_hash.insert(track.id);
+    }
+
+    // Create hash and vec of 'from_ids'
+    let mut hasher = DefaultHasher::new();
```

Comment:
There is a problem with using a hash to compare changes as this assumes static ordering between items in a playlist. For example, a hash of `A, B, C` will be different than `C, A, B`.